### ISSUE TITLE
V 0.1.21 - bugs

### ISSUE THREAD

**Issue by ManPython** [vlang/v#2197]
v up from V 0.1.20 to new was fail, no v.exe

After git clone and make was ok, but simple examples not working coretly.
Runing v.exe doing nothing like other compiled .exe

Want sugesting some proxy method for versions of v like v.exe proxing to V 0.1.21 .exe or V 0.1.20.exe by config.ini and can be used version maybe.

`rebuild from source (twice, in case of C definitions changes)`
v compilation is longer now . It's finally forever ?

**Comment by lutherwenxu** [comment]
Can you please be more specific about the problems you are facing? What were you trying to do? What did you expect would happen? What actually happened?

**Comment by ManPython** [comment]
Does v up offer some debug mode? If no then hard to explain.
Eg. hello_world.v not printing hello like in previous versions.
I expect to compile v in v version but the .exe not producing corect .exe files.

**Comment by spytheman** [comment]
You can debug programs with gdb/msvc. v supports passing -debug so that the produced executables have more information in them like line numbers/symbols.

**Comment by spytheman** [comment]
Please specify your OS and C compiler. On windows, try to use mingw-w64, or msvc (NB: the 32bit mingw32 is NOT supported).

**Comment by ManPython** [comment]
W7x64 sp1
mingw-w64\x86_64-8.1.0-release-win32-sjlj-rt_v6-rev0\mingw64\bin

**Comment by ManPython** [comment]
v-0.1.22 this same problems.. nothing works.. 
Strange.. only v-0.1.19 works ok.

**Comment by ManPython** [comment]
Can someone put here compiled v.exe as from v-0.1.22?

**Comment by gslicer** [comment]
Here is my updated V for Windows for you to check:
[V 0.1.22 dbaa37e.zip](https://github.com/vlang/v/files/3896898/V.0.1.22.dbaa37e.zip)

**Comment by ManPython** [comment]
Well.. it's not working. The hello_world.exe is compiled but not printing nothing on cmd.
When runing v.exe no any console with >> or version info just `_` blinking.
Windows-7-6.1.7601-SP1 64 Bits, don't know whats going on.

**Comment by ManPython** [comment]
And not working on difrent laptop and W10. Very strange.

**Comment by gslicer** [comment]
@ManPython could you try the official package please?
[V for Windows](https://github.com/vlang/v/releases/latest/download/v_windows.zip)

**Comment by ManPython** [comment]
@gslicer stoped worked by windows alert. Not working.

**Comment by afonichev** [comment]
Win10 (x64)

[0.1.23.1](https://github.com/vlang/v/releases/tag/0.1.23.1)

![image](https://user-images.githubusercontent.com/15726637/70229139-ad4f9b80-1766-11ea-8ebb-9350af61f7a2.png)

**Comment by ManPython** [comment]
@medvednikov any progress here?

**Comment by gslicer** [comment]
@ManPython it seems like your windows is somehow irregular - did you install an original .iso from Microsoft and scanned for Viruses/TrojanHorses?

**Comment by ManPython** [comment]
@gslicer I don't belive your opinion coz my friendly coders tested this same and me on other computers and OS like w7, w10. Soo.. on Windows are some problems. Notice that first version working.. soo.. it's not lok like OS problem..

**Comment by ManPython** [comment]
https://github.com/vlang/v/releases/tag/0.1.24
Still this same case.

**Comment by vitalyster** [comment]
@ManPython prbuilt version on vlang.io should work now.
Note, if you does not have MSVC installed then you need to start V from MinGW command prompt (with `gcc` in PATH)

**Comment by ManPython** [comment]
There is no cmake on 0.1.24 so not sure it's actual to compile by MinGW especialy that previous vestions show some problems around this. Right.. don't have MSVC..  but even that v.exe exist.. should compile hello_world.v correctly (?) but nod doing this like in previous versions.
@afonichev  can you confirm this same?

Should I expect v2.0 as working? I hope.. if not I probably leave interest for V. Something was broken deptly.

**Comment by vitalyster** [comment]
@ManPython can you please show actual error output from latest version?

**Comment by vitalyster** [comment]
@ManPython v_windows.zip - it is compiled version (no make.bat)
Source code (.zip) - is the source with make.bat

**Comment by medvednikov** [comment]
This was fixed in 0.1.24.

**Comment by ManPython** [comment]
@medvednikov 
> This was fixed in 0.1.24.

I tested now 0.1.24 coz subscribed new releases and always testing and weiting for working version.
I compiled v by MinGW but this same.. In both option when run v.exe no any console just blinking `_`

@vitalyster 
There is no any error output in case of compile like `v run hello_world.v -debug` and when runing nothing is happen (by exe or in cmd).

**Comment by vitalyster** [comment]
@ManPython I'm seeing the same problem on Win7, but not on Win10, can you test on Win10?

**Comment by vitalyster** [comment]
Issue with wrong C runtime library used by MinGW-W64 on Windows 7.
Just tested with llvm-mingw and everything works correctly. You can try yourself - https://github.com/mstorsjo/llvm-mingw/releases/tag/20191230

**Comment by ManPython** [comment]
@vitalyster 
I tested w10. The compiled hello_world.exe works as orginal v.exe output and mingw (0.1.24 - previous not).
But runing v.. 
Orginal v.exe
`print_backtrace_skipping_top_frames is not implemented on this platform for now...`
MinGW v.exe
```
V panic: V tool "C:\v-0.1.24_\tools\vrepl.v" could not be compiled
V error: Cannot find MSVC on this OS
TODO: print_backtrace_skipping_top_frames_mingw(2)
print_backtrace_skipping_top_frames is not implemented on this platform for now...
```

Soo.. I need compile v for w7 by llvm-mingw?

**Comment by vitalyster** [comment]
@ManPython you need to have `gcc` in PATH when running V

**Comment by ManPython** [comment]
@vitalyster I have..? I compiled v by cmake by MinGW.. soo?

**Comment by vitalyster** [comment]
@ManPython you can type `gcc` in the same command prompt window to make sure you have it.

**Comment by ManPython** [comment]
Sure.. I know, but this is not related issue I think.

```
C:\>gcc --version
gcc (x86_64-win32-sjlj-rev0, Built by MinGW-W64 project) 8.1.0
Copyright (C) 2018 Free Software Foundation, Inc.
This is free software; see the source for copying conditions.  There is NO
warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
```

**Comment by vitalyster** [comment]
Now `cd to/v/directory` and `v` ?

**Comment by ManPython** [comment]
I have v in path soo I can call v from any directory but after v enter sound like OS working and trying run v.exe but nothing happen. 
HVR.. https://github.com/vlang/v/issues/2197#issuecomment-572529642
On w7 hello_world.exe is compiled just this file and app not working on w7. Good that you sugest test this on w10 coz I was sure that this all was broken - now know that thus us w7 issue, but this not solving problem and on w10 are other problems. Sooo.. it's all far from expectations. Waiting for stable version.

**Comment by vitalyster** [comment]
> I can call v from any directory

I'm not sure it is working and we can not see exact reason of error because `print_backtrace` is really not implemented for MinGW builds.
That is why you need to test from v directory, I just tested on Win7 and it works.
 
> nothing happen.

I already explain it before - use llvm-mingw toolchain and it will work on Win7 and print to console correctly.

> w10 are other problems

Which problems you have on Win10?

> Waiting for stable version

You can wait it forever if you will not report bugs.

**Comment by ManPython** [comment]
> > w10 are other problems
> 
> Which problems you have on Win10?
No console output in cmd or when v is runed as first version of v.
And if the first version of v works like I described on 1st post minds its possible probably do this friendly for all OS's?

**Comment by vitalyster** [comment]
@ManPython "No console output" on Win10? Which exact version (Windows 10 1709 is minimum recommended as described here - https://github.com/vlang/v/wiki/Installing-a-C-compiler-on-Windows). Did you build on Win10 or copy v.exe from Win7? Does it work with vlang.io version?

**Comment by ManPython** [comment]
> @ManPython "No console output" on Win10?
Right. Oh.. I missed the requiments updated on 3 dec about w10 as 1709.  Must check next time version, it's updated automaticly, but not sure on this moment. I used copy that I compiled on w7 and download from https://github.com/vlang/v/releases/tag/0.1.24 (as vlang.io version). Both not working.

**Comment by vitalyster** [comment]
@ManPython I can not reproduce it, so you are doing something wrong and can not describe things correctly. Few messages before you say it works on Windows 10, now it is "not working". I do not want to waste my time with your poor reports anymore.

**Comment by ManPython** [comment]
Maybe post was not clear. The hello_world.exe give output, but v.exe (as compiler) not doing print as console `>>> ` version info etc.. like in early versions of v.exe when I tested and this was.

**Comment by ManPython** [comment]
@medvednikov solved in v-0.1.29
